python-checks/src/main/java/org/sonar/python/checks/cdk/S3BucketVersioningCheck.java

```diff
@@ -0,0 +1,98 @@
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.python.checks.cdk;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import org.sonar.check.Rule;
+import org.sonar.plugins.python.api.PythonSubscriptionCheck;
+import org.sonar.plugins.python.api.SubscriptionContext;
+import org.sonar.plugins.python.api.tree.Argument;
+import org.sonar.plugins.python.api.tree.CallExpression;
+import org.sonar.plugins.python.api.tree.Expression;
+import org.sonar.plugins.python.api.tree.Name;
+import org.sonar.plugins.python.api.tree.RegularArgument;
+import org.sonar.plugins.python.api.tree.Token;
+import org.sonar.plugins.python.api.tree.Tree;
+import org.sonar.python.checks.Expressions;
+
+@Rule(key = "S6252")
+public class S3BucketVersioningCheck extends PythonSubscriptionCheck {
+
+  public static final String MESSAGE = "Make sure an unversioned S3 bucket is safe here.";
+  public static final String MESSAGE_OMITTING = "Omitting the \"versioned\" argument disables S3 bucket versioning. Make sure it is safe here.";
+  public static final String MESSAGE_SECONDARY = "Propagated setting.";
+
+  @Override
+  public void initialize(Context context) {
+    context.registerSyntaxNodeConsumer(Tree.Kind.CALL_EXPR, this::visitNode);
+  }
+
+  public void visitNode(SubscriptionContext ctx) {
+    CallExpression node = (CallExpression) ctx.syntaxNode();
+    Optional.ofNullable(node.calleeSymbol())
+      .filter(nodeSymbol -> ("aws_cdk.aws_s3.Bucket".equals(nodeSymbol.fullyQualifiedName())))
+      .ifPresent(nodeSymbol -> {
+        Optional<RegularArgument> version = getVersionArgument(node.arguments());
+        if (version.isPresent()) {
+          version.map(regArg -> treesOfFalseExpression(regArg.expression()))
+            .filter(trees -> !trees.isEmpty())
+            .ifPresent(trees -> {
+              PreciseIssue issue = ctx.addIssue(trees.get(0).parent(), MESSAGE);
+              trees.stream().skip(1).forEach(expression -> issue.secondary(expression.parent(), MESSAGE_SECONDARY));
+            });
+        } else {
+          ctx.addIssue(node.callee(), MESSAGE_OMITTING);
+        }
+      });
+  }
+
+  private static Optional<RegularArgument> getVersionArgument(List<Argument> args) {
+    return args.stream()
+      .map(RegularArgument.class::cast)
+      .filter(a -> a.keywordArgument() != null)
+      .filter(a -> "versioned".equals(a.keywordArgument().name()))
+      .findAny();
+  }
+
+  private static List<Tree> treesOfFalseExpression(Expression expression) {
```

Comment:
```suggestion
  private static List<Tree> propagatedSenitiveExpressions(Expression expression) {
```

Reply:
Yes that's a better name, I've changed it